packages/sfui/shared/focusVisibleManager/focusVisibleManager.ts

```diff
@@ -0,0 +1,172 @@
+import { isAndroid, isMac } from '../utils';
+import { FocusHandler, FocusHandlerEvent, FocusModality } from './types';
+
+// This code has been implemented based on @react-aria useFocusVisible
+// https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/interactions/src/useFocusVisible.ts
+
+// Keyboards, Assistive Technologies, and element.click() all produce a "virtual"
+// click event. This is a method of inferring such clicks. Every browser except
+// IE 11 only sets a zero value of "detail" for click events that are "virtual".
+// However, IE 11 uses a zero value for all click events. For IE 11 we rely on
+// the quirk that it produces click events that are of type PointerEvent, and
+// where only the "virtual" click lacks a pointerType field.
+
+function isVirtualClick(event: MouseEvent | PointerEvent): boolean {
+  // JAWS/NVDA with Firefox.
+  if ((event as any).mozInputSource === 0 && event.isTrusted) {
```

Comment:
I'm wondering if this is valid way of checking is its virtual click, kind of sketchy IMHO :D 
https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/mozInputSource

`Non-standard: This feature is non-standard and is not on a standards track. Do not use it on production sites facing the Web: it will not work for every user. There may also be large incompatibilities between implementations and the behavior may change in the future.`

But I guess adobe tested it :D 🤷

Reply:
I'll look into it. Thanks for catching it!